### ISSUE TITLE
Transformer layer实现有问题，测试不通过，我实现完成了Transformer layer，并通过了测试。但是Transformer model实现有问题，代码如下，想请教下，如何改正，谢谢。

### ISSUE THREAD

**Issue by piood** [wmhst7/CMU-10-714-DL-Sys#1]
我实现完成了Transformer layer，并通过了测试
class TransformerLayer(Module):

    def __init__(
        self,
        q_features: int,
        num_head: int,
        dim_head: int,
        hidden_size: int,
        *,
        dropout = 0.,
        causal = True,
        device = None,
        dtype = "float32",
    ):

        super().__init__()

        self.device = device
        self.dtype = dtype

        ### BEGIN YOUR SOLUTION
        self.attention = AttentionLayer(
            q_features, num_head, dim_head,
            dropout=dropout, causal=causal,
            device=device, dtype=dtype
        )
        
        self.norm1 = LayerNorm1d(q_features, device=device, dtype=dtype)
        self.norm2 = LayerNorm1d(q_features, device=device, dtype=dtype)
        
        self.linear1 = Linear(q_features, hidden_size, device=device, dtype=dtype)
        self.linear2 = Linear(hidden_size, q_features, device=device, dtype=dtype)
        
        self.relu = ReLU()
        
        self.dropout = Dropout(dropout)
            
        ### END YOUR SOLUTION

    def forward(
        self,
        x
    ):
        """
        The forward function of a Transformer Layer.
        Input: the hidden states from previous layers `x` with shape (batch_size, seq_len, x_dim)
        Ouput: the hidden states after the Transformer Layer `x` with shape (batch_size, seq_len, x_dim)
        """

        batch_size, seq_len, x_dim = x.shape

        ### BEGIN YOUR SOLUTION
        residual = x
        
        #x = self.norm1(x)
        x = self.attention(x)
        x = self.dropout(x)
        x = x + residual
        
        residual = x
        x = self.norm2(x)
        x = self.linear1(x)
        x = self.relu(x)
        x = self.dropout(x)
        x = self.linear2(x)
        x = self.dropout(x)
        x = x + residual
        ### END YOUR SOLUTION

        return x

Transformer model输出有问题，未通过测试。
class Transformer(Module):

    def __init__(
        self,
        embedding_size: int,
        hidden_size: int,
        num_layers: int, 
        *,
        num_head: int = 8,
        dim_head: int = 32,
        dropout = 0.,
        causal = True,
        device = None,
        dtype = "float32",
        batch_first = False,
        sequence_len = 2048
    ):

        super().__init__()

        self.device = device
        self.dtype = dtype
        self.batch_first = batch_first

        ### BEGIN YOUR SOLUTION
        # 定义 Transformer layers
        self.layers = Sequential(
            *[TransformerLayer(embedding_size, num_head, dim_head, hidden_size,
                               dropout=dropout, causal=causal,
                               device=device, dtype=dtype) for _ in range(num_layers)]
        )

        # 定义位置编码
        self.pos_embedding = Embedding(sequence_len, embedding_size, device=device, dtype=dtype)
        ### END YOUR SOLUTION

    def forward(
        self,
        x, h=None
    ):

        if not self.batch_first:
            x = ops.transpose(x, axes=(0, 1))

        ### BEGIN YOUR SOLUTION
        # 获取输入序列的长度
        seq_len, batch_size, _ = x.shape

        # 创建位置序列
        positions = np.arange(seq_len).reshape(1, seq_len)
        print("positions", positions)
        positions = Tensor(positions, device=self.device, dtype=self.dtype, requires_grad=False)
        positions = positions.broadcast_to((batch_size, seq_len))
        positions = ops.transpose(positions, axes=(1, 0))

        # 获取位置编码
        pos_emb = self.pos_embedding(positions)
        print("pos_emb", pos_emb)
        # 将位置编码添加到输入嵌入中
        x = x + pos_emb

        # 通过 Transformer layers
        x = self.layers(x)
        ### END YOUR SOLUTION

        if not self.batch_first:
            x = ops.transpose(x, axes=(0, 1))

        return x, init.zeros_like(x)
